packages/client/src/NetworkNodeFacade.ts

```diff
@@ -196,45 +212,58 @@ export class NetworkNodeFacade {
      * but will be sync in case that node is already started.
      * Zalgo intentional. See below.
      */
-    publishToNode(streamMessage: StreamMessage): void | Promise<void> {
+    async publishToNode(streamMessage: StreamMessage): Promise<void> {
         // NOTE: function is intentionally not async for performance reasons.
         // Will call cachedNode.publish immediately if cachedNode is set.
         // Otherwise will wait for node to start.
         this.destroySignal.assertNotDestroyed()
         if (this.isStarting()) {
             // use .then instead of async/await so
             // this.cachedNode.publish call can be sync
-            return this.startNodeTask().then((node) => {
-                return node.publish(streamMessage)
-            })
+            return this.startNodeTask().then((node) =>
+                node.publish(streamMessage)
```

Comment:
Should we `await` here? (also note comment at line 222, maybe not relevant anymore?)